### PR TITLE
gh-124855: Fix jit perf (and lejit)

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
JIT builds are currently failing on main. This supersedes https://github.com/python/cpython/pull/126163. This wasn't caught because we don't run the JIT tests on `pylifecycle.c`

* Issue: gh-124855

### CODE REVIEW

**Comment by pablogsal** [comment]
I think we need to add skips to the tests in the Perf trampoline when the jit is activated

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
I have pushed a bunch of fixes to ensure we run the tests that check for anything perf without the JIT as we do in other places

File: Lib/test/test_embed.py [hunk 1]
```diff
@@ -1768,6 +1768,8 @@ def test_init_set_config(self):
         self.check_all_configs("test_init_set_config", config,
                                api=API_ISOLATED)
 
+    @unittest.skipIf("-D_Py_JIT" in (sysconfig.get_config_var('PY_CORE_CFLAGS') or ''),
```
Comment (pablogsal) [inline Lib/test/test_embed.py]: This fails on windows for some reason

Comment (pablogsal) [reply Lib/test/test_embed.py]: I think with my fix we can actually run the test so I am switching to that

Comment (Fidget-Spinner) [reply Lib/test/test_embed.py]: Hmm I'm abit stumped. No clue why your fix isn't working :(.

Comment (pablogsal) [reply Lib/test/test_embed.py]: It's because the test is running on isolated mode and therefore it ignores the environment flags

Comment (pablogsal) [reply Lib/test/test_embed.py]: I pushed a fix that should do the trick

**Comment by Fidget-Spinner** [comment]
Still failing on Windows... not sure why

**Comment by Eclips4** [comment]
Okay, I finally set up my Windows environment and it seems that there is no `PY_CORE_CFLAGS`. Perhaps we can't currently determine if JIT is enabled. Maybe it's time to add a `PY_JIT` variable to `pyconfig.h`?

**Comment by pablogsal** [comment]
Maybe we can land a version of this that just ignores warning lines in stderr and then undo that hack when we have JIT detection on windows

**Comment by mpage** [comment]
> Maybe we can land a version of this that just ignores warning lines in stderr and then undo that hack when we have JIT detection on windows.
> 
> This way we can move on and avoid having to revert

I put up https://github.com/python/cpython/pull/126302 as a possible work-around.

**Comment by Eclips4** [comment]
Closing it due to merge of #126302. Thank you Pablo and Ken.

**Pull request closed without merging**